hopper/src/main/java/org/atomhopper/abdera/parser/UnifiedParser.java

```diff
@@ -0,0 +1,278 @@
+package org.atomhopper.abdera.parser;
+
+import org.apache.abdera.model.Document;
+import org.apache.abdera.model.Element;
+import org.apache.abdera.parser.ParseException;
+import org.apache.abdera.parser.Parser;
+import org.apache.abdera.parser.ParserOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import javax.xml.stream.XMLStreamReader;
+import java.io.InputStream;
+import java.io.Reader;
+import java.nio.channels.ReadableByteChannel;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+@Component
+public class UnifiedParser implements Parser {
+
+    private final Parser xmlParser;
+    private final JsonAtomParser jsonParser;
+    private ParserOptions defaultParserOptions;
+
+    static Logger LOG = LoggerFactory.getLogger(UnifiedParser.class);
+
+    public UnifiedParser(Parser xmlParser) {
+        if (xmlParser == null) {
+            LOG.error("XML parser is null in constructor");
+            throw new IllegalArgumentException("XML parser cannot be null");
+        }
+        this.xmlParser = xmlParser;
+        this.jsonParser = new JsonAtomParser();
+        this.defaultParserOptions = xmlParser.getDefaultParserOptions();
+        LOG.info("UnifiedParser is initialized with XMLParser: {}", xmlParser.getClass().getSimpleName());
+    }
+
+    private static final Set<String> JSON_TYPES = Collections.unmodifiableSet(
+            new HashSet<>(Arrays.asList(
+                    ContentType.APPLICATION_JSON.getValue(),
+                    ContentType.TEXT_JSON.getValue(),
+                    ContentType.VENDOR_JSON.getValue()
+            ))
+    );
+
+    private static final Set<String> XML_TYPES = Collections.unmodifiableSet(
+            new HashSet<>(Arrays.asList(
+                    ContentType.APPLICATION_XML.getValue(),
+                    ContentType.TEXT_XML.getValue(),
+                    ContentType.APPLICATION_ATOM_XML.getValue()
+            ))
+    );
+    private boolean isJsonContent(String contentType) {
+        if (contentType == null) return false;
+        String normalized = contentType.toLowerCase().trim();
+        return JSON_TYPES.stream().anyMatch(normalized::contains)
+                || normalized.contains("+json");
+    }
+
+    private boolean isXmlContent(String contentType) {
+        if (contentType == null) return false;
+        String normalized = contentType.toLowerCase().trim();
+        return XML_TYPES.stream().anyMatch(normalized::contains)
+                || normalized.contains("+xml");
+    }
```

Comment:
These contains checks can approve unsupported content types. We also need to be able to trim out encoding if it gets appended to the content type header.